lib/core/utils/cache_interceptor.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:project_kepler/core/global.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'dart:convert';
+
+class CacheInterceptor extends Interceptor {
+  final SharedPreferences sharedPreferences;
+
+  CacheInterceptor(this.sharedPreferences);
+
+  @override
+  void onRequest(
+      RequestOptions options, RequestInterceptorHandler handler) async {
+    final key = options.uri.toString();
+    final cachedData = sharedPreferences.getString(key);
+
+    if (cachedData != null) {
+      final data = jsonDecode(cachedData);
+
+      final expiryDate = DateTime.tryParse(data['expiryDate']);
+
+      logger.d(
+          "Cache expiry in  ${expiryDate?.difference(DateTime.now()).inSeconds} seconds");
```

Comment:
Logging with potentially null `expiryDate` could result in printing 'null seconds'. It's better to handle the null case explicitly for clearer logs.

```diff
-          "Cache expiry in  ${expiryDate?.difference(DateTime.now()).inSeconds} seconds"
+          "Cache expiry in  ${expiryDate != null ? expiryDate.difference(DateTime.now()).inSeconds : 'unknown'} seconds"
```